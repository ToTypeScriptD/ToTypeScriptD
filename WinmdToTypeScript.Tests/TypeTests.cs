using ApprovalTests;
using NUnit.Framework;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WinmdToTypeScript.Tests
{

    public class TypeTests : TestBase
    {
        [Test]
        public void EnumType()
        {
            var result = GetNativeType("SampleEnum").ToTypeScript();
            Approvals.Verify(result);
        }

        [Test]
        public void EnumTypeWithExplicitValues()
        {
            var result = GetNativeType("SampleEnumNumbered").ToTypeScript();
            Approvals.Verify(result);
        }

        [Test]
        public void ClassWithEventHandler()
        {
            var result = GetNativeType("ClassWithEventHandler").ToTypeScript();
            Approvals.Verify(result);
        }

        [Test]
        public void FullSampleAssembly()
        {
            var result = WinmdToTypeScript.Render.FullAssembly(base.NativeComponentPath);
            Approvals.Verify(result);
        }

        [Test]
        public void WindowsStorageClass()
        {
            var result = GetWinNativeType("Windows.Storage.StorageFile").ToTypeScript();
            Approvals.Verify(result);
        }
        [Test]
        public void WindowsSystemUserProfileGlobalizationPreferencesClass()
        {
            var result = GetWinNativeType("Windows.System.UserProfile.GlobalizationPreferences").ToTypeScript();
            Approvals.Verify(result);
        }
        
        //[Test]
        //public void FullWindowsAssembly()
        //{
        //    var file = @"C:\Windows\System32\WinMetadata\Windows.Foundation.winmd";
        //    var result = WinmdToTypeScript.Render.FullAssembly(file);
        //    Approvals.Verify(result);
        //}
    }
}
