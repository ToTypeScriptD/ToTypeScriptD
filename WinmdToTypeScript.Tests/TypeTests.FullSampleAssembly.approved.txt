declare module WinmdToTypeScript.Native {

    export class Class1 implements WinmdToTypeScript.Native.__IClass1PublicNonVirtuals {
        constructor();
    }

    export class ClassWithEventHandler implements WinmdToTypeScript.Native.__IClassWithEventHandlerPublicNonVirtuals, WinmdToTypeScript.Native.__IClassWithEventHandlerProtectedNonVirtuals {
        addEventListener(type: string, listener: EventListener): void;
        removeEventListener(type: string, listener: EventListener): void;
        onsomethinghappened(ev: any);
        constructor();
        doSomething(): void;
    }

    enum SampleEnum {
        a,
        b,
        c,
        d
    }

    enum SampleEnumNumbered {
        a,
        b,
        c,
        d
    }

    export class SomethingHappenedEventHandler {
        constructor(__param0: System.Object, __param1: System.IntPtr);
        invoke(__param0: WinmdToTypeScript.Native.ClassWithEventHandler, __param1: string): void;
    }

}
